hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/CopyOnWriteRollbackActionExecutor.java

```diff
@@ -67,11 +68,23 @@ protected List<HoodieRollbackStat> executeRollback(HoodieRollbackPlan hoodieRoll
 
     if (instantToRollback.isCompleted()) {
       LOG.info("Unpublishing instant " + instantToRollback);
+      table.getMetaClient().getTableFormat().rollback(instantToRollback, table.getContext(), table.getMetaClient(), table.getViewManager());
+      // Revert the completed instant to inflight in native format.
       resolvedInstant = activeTimeline.revertToInflight(instantToRollback);
       // reload meta-client to reflect latest timeline status
       table.getMetaClient().reloadActiveTimeline();
     }
 
+    // If instant is inflight but marked as completed in native format, delete the completed instant from storage.
```

Comment:
nts: to review closely

Reply:
yeah, let's avoid to listing the timeline multiple times for reqular workflow when the external table format is not there.